mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -277,9 +277,13 @@ pub async fn tear_down_bot(ctx: MmArc) {
     if let TradingBotState::Stopped(ref mut stopped_state) = *state {
         let nb_orders = cancel_pending_orders(&ctx, &stopped_state.trading_bot_cfg.clone()).await;
         let dispatcher_ctx = DispatcherContext::from_ctx(&ctx).unwrap();
-        let dispatcher = dispatcher_ctx.dispatcher.lock().await;
         let event: TradingBotEvent = TradingBotStopped { nb_orders }.into();
-        dispatcher.dispatch_async(ctx.clone(), event.into()).await;
+        dispatcher_ctx
+            .dispatcher
+            .read()
+            .await
+            .dispatch_async(ctx.clone(), event.into())
+            .await;
```

Comment:
What do you think about adding a static function `dispatch_lp_event` that will look like:
```rust
pub fn dispatch_lp_event(ctx: MmArc, event: LpEvents) -> Result<(), MmError<DispatcherError>> {
  let dispatcher_ctx = DispatcherContext::from_ctx(&ctx).or_mm_error(|| Error::MmCtxNotAvailable)?;
  dispatcher_ctx.dispatcher.read().await.dispatch_async(ctx, event).await;
  Ok()
}
```

I think we can add it because we're supposed to produce events from a lot different places in the future, and it will reduce amount of code.

Reply:
Sounds like a good idea, will do!

Reply:
Can you please replace `dispatcher_ctx.dispatcher.read().await.dispatch_async(ctx.clone(), event.into()).await;` with `dispatch_lp_event().await`?